pyproject.toml

```diff
@@ -9,39 +9,86 @@ requires = [
 [tool.ruff]
 target-version = "py39"
 show-fixes = true
+show-source = true
+# Find info about ruff rules at: https://docs.astral.sh/ruff/rules
 extend-select = [
+  "ARG", # flake8-unused-arguments
+  "B", # flake8-bugbear
+  "BLE", # flake8-blind-except
+  "C4", # flake8-comprehensions
+  "C90", # mccabe
+  "COM818", # trailing-comma-on-bare-tuple
+  "FBT003", # flake8-boolean-trap
+  "FLY", # flynt
+  "I", # isort
+  "ICN", # flake8-import-conventions
+  "INP", # flake8-no-pep420
+  "INT", # flake8-gettext
+  "ISC", # flake8-implicit-str-concat
+  "N", # pep8-naming
+  "NPY", # numpy-deprecated-type-alias
+  "PD", # pandas-vet
+  "PERF", # perflint
+  "PGH", # pygrep-hooks
+  "PIE", # flake8-pie
+  "PLC", # pylint convention
+  "PLE", # pylint errors
+  "PLW", # pylint warnings
+  "PT", # flake8-pytest-style
+  "PTH", # flake8-use-pathlib
+  "PYI", # flake8-pyi
+  "RSE", # flake8-raise
+  "RUF005",# collection-literal-concatenation
+  "RUF006", # asyncio-dangling-task
+  "RUF007", # pairwise-over-zipped
+  "RUF008", # mutable-dataclass-default
+  "RUF009", # function-call-in-dataclass-default-argument
+  "RUF010", # explicit-f-string-type-conversion
+  "RUF011", # static-key-dict-comprehension
+  "RUF013", # implicit-optional
+  "RUF015", # unnecessary-iterable-allocation-for-first-element
+  "RUF016", # invalid-index-type
+  "RUF100", # unused-noqa
+  "RUF200", # invalid-pyproject-toml
+  "S", # flake8-bandit
+  "SIM", # flake8-simplify
+  "TCH", # flake8-type-checking
+  "TID", # flake8-tidy-imports
+  "TRY", # tryceratops
   "UP", # pyupgrade
+  "W", # pycodestyle warnings
 ]
 ignore = [
-  "E501",
+  "E501", # line-too-long
+  "N802", # invalid-function-name
+  "N803", # invalid-argument-name
+  "N806", # non-lowercase-variable-in-function
+  "N816", # mixed-case-variable-in-global-scope
+  "S101", # asserts
+  "SIM108", # if-else-block-instead-of-if-exp
+  "TRY003", # raise-vanilla-args
 ]
 extend-exclude = [
   ".jupyter",
   "__pycache__",
   "_build",
   "_dev",
 ]
+namespace-packages = [".github/workflows", "docs"]
+
+
+[tool.ruff.per-file-ignores]
+"__init__.py" = ["E402", "F401", "F402", "F403"]  # ignore import errors
+"docs/notebooks/computing_functions/temperature_response.ipynb" = ["A001"]  # filter variable shadows Python builtin
 
 [tool.ruff.flake8-import-conventions.aliases]
 "astropy.units" = "u"
 "matplotlib.pyplot" = "plt"
 numpy = "np"
 pandas = "pd"
 
-[tool.ruff.per-file-ignores]
-"__init__.py" = ["E402", "F401", "F402", "F403"]  # ignore import errors
-
-[tool.isort]
-line_length = 88
-wrap_length = 80
-sections = ["FUTURE", "STDLIB", "FIRSTPARTY", "LOCALFOLDER"]
-known_first_party = ["xrtpy"]
-default_section = "STDLIB"
-multi_line_output = 3
-use_parentheses = true
-include_trailing_comma = true
-force_alphabetical_sort_within_sections = true
-lines_between_types = 1
+[tool.ruff.mccabe]
+max-complexity = 12
```

Comment:
The `mccabe` rule set corresponds to [cyclomatic complexity](https://en.wikipedia.org/wiki/Cyclomatic_complexity) (also known as McCabe complexity).  Essentially it's saying that functions with lots of `for` and nested `if` statements should be broken up into smaller, less complex functions.